src/ln/channelmanager.rs

```diff
@@ -1656,6 +1654,20 @@ impl ChannelMessageHandler for ChannelManager {
 				Some(id) => id.clone(),
 			};
 			let chan = channel_state.by_id.get_mut(&forwarding_id).unwrap();
+			let fee = 0;
+			if next_hop_data.hmac != [0;32] {
+				let fee = chan.get_our_fee_base_msat(&*self.fee_estimator) + (pending_forward_info.amt_to_forward * self.fee_proportional_millionths as u64 / 1000000) as u32;
```

Comment:
Because you have a "let" here, fee is always 0 in the next if statement. Also, the pending_forward_info.onion_packet.is_some() check above is duplicative with this - if next_hop_data.hmac is 0s onion_packet is always None.